packages/react-router/__tests__/descendant-routes-splat-matching-test.tsx

```diff
@@ -57,5 +57,72 @@ describe("Descendant <Routes> splat matching", () => {
         </div>
       `);
     });
+    it("works with paths beginning with special characters", () => {
```

Comment:
Should I add tests here for the other new characters?
Also, should I add a test for not matching `%` when it's not an escape for an encoded entity?

Reply:
More exhaustive tests are always a good thing. 

You can extract the bulk of this test out to a function that takes the initialEntries as a parameter and just iterate over a few different forms. That will let you test more paths without having to write a ton of extra code.